pkg/apis/operator/v1alpha1/apiserver_types.go

```diff
@@ -18,84 +18,11 @@ package v1alpha1
-	// "https://accounts.google.com" or "https://login.salesforce.com".
-	//
-	// The provider must implement configuration discovery.
-	// See: https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig
-	IssuerURL string `json:"issuerURL"`
-
-	// ClientID the JWT must be issued for, the "sub" field. This plugin only trusts a single
-	// client to ensure the plugin can be used with public providers.
-	//
-	// The plugin supports the "authorized party" OpenID Connect claim, which allows
-	// specialized providers to issue tokens to a client for a different client.
-	// See: https://openid.net/specs/openid-connect-core-1_0.html#IDToken
-	ClientID string `json:"clientID"`
-
-	// APIAudiences are the audiences that the API server identitifes as. The
-	// (API audiences unioned with the ClientIDs) should have a non-empty
-	// intersection with the request's target audience. This preserves the
-	// behavior of the OIDC authenticator pre-introduction of API audiences.
-	// +optional
-	APIAudiences authenticator.Audiences `json:"apiAudiences,omitempty"`
-
-	// CertificateAuthority references the secret containing issuer's CA in a PEM encoded root certificate of the provider.
-	CertificateAuthority ObjectReference `json:"certificateAuthority"`
-
-	// UsernameClaim is the JWT field to use as the user's username.
-	// +kubebuilder:default=sub
-	// +optional
-	UsernameClaim string `json:"usernameClaim"`
-
-	// UsernamePrefix, if specified, causes claims mapping to username to be prefix with
-	// the provided value. A value "oidc:" would result in usernames like "oidc:john".
-	// +optional
-	UsernamePrefix string `json:"usernamePrefix,omitempty"`
-
-	// GroupsClaim, if specified, causes the OIDCAuthenticator to try to populate the user's
-	// groups with an ID Token field. If the GroupsClaim field is present in an ID Token the value
-	// must be a string or list of strings.
-	// +optional
-	GroupsClaim string `json:"groupsClaim,omitempty"`
-
-	// GroupsPrefix, if specified, causes claims mapping to group names to be prefixed with the
-	// value. A value "oidc:" would result in groups like "oidc:engineering" and "oidc:marketing".
-	// +optional
-	GroupsPrefix string `json:"groupsPrefix,omitempty"`
-
-	// SupportedSigningAlgs sets the accepted set of JOSE signing algorithms that
-	// can be used by the provider to sign tokens.
-	//
-	// https://tools.ietf.org/html/rfc7518#section-3.1
-	//
-	// This value defaults to RS256, the value recommended by the OpenID Connect
-	// spec:
-	//
-	// https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation
-	// +kubebuilder:default=RS256;
-	SupportedSigningAlgs []string `json:"supportedSigningAlgs,omitempty"`
-
-	// RequiredClaims, if specified, causes the OIDCAuthenticator to verify that all the
-	// required claims key value pairs are present in the ID Token.
-	// +optional
-	RequiredClaims map[string]string `json:"requiredClaims,omitempty"`
+	API APIServerConfig `json:"api,omitempty"`
```

Comment:
This does not need to be nested, please leave these fields in `APIServerSpec` and reference the `APIServerSpec` as `api` key from the `KubeCarrierSpec`.

e.g.

```golang
type KubeCarrierSpec struct {
	// +optional
	API APIServerSpec `json:"api,omitempty"`
}
```

I would like to avoid calling objects `*Config` - as all these APIs are declarative, they are all "Configs" or Specifications and `Spec` is way more common in k8s objects.

Reply:
Yeah, make sense, will do that

Reply:
done